custom_components/samsungtv_custom/media_player.py

```diff
@@ -185,17 +190,33 @@ def __init__(self, host, port, name, timeout, mac, uuid, sourcelist, protocol):
             "port": port,
             "host": host,
             "timeout": timeout,
+            "token": token,
         }
         self._sourcelist = sourcelist
         
-        if self._config["port"] in (8001, 8002):
+        if self._token != "":
+            self._config["method"] = "encrypted"
+        elif self._config["port"] in (8001, 8002):
             self._config["method"] = "websocket"
         else:
             self._config["method"] = "legacy"
 
     def update(self):
         """Update state of device."""
-        self.send_key("KEY")
+        if self._config["method"] == "encrypted" and self._config["port"] == 8001:
+            try:
+                r = requests.get(
+                    "http://{}:8001/api/v2/".format(self._config["host"]), timeout=2
```

Comment:
Is there a reason for hardcoding port?

Reply:
Well yes, because for these models 8001 is http. So if there are scenarios using 8002, we would have to use https. But afaik these would not use that encrypted remote thing. But please correct me if I am wrong.